core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -206,7 +213,10 @@ public synchronized void connect() throws AlluxioStatusException {
         beforeConnect();
         LOG.info("Alluxio client (version {}) is trying to connect with {} @ {}",
             RuntimeConstants.VERSION, getServiceName(), mAddress);
-        mChannel = GrpcChannelBuilder.forAddress(mAddress).setSubject(mParentSubject).build();
+        mChannel = GrpcChannelBuilder
+            .newBuilder(mAddress, mContext.getConf())
```

Comment:
why doesn't this just take a `ClientContext`?

Reply:
Because the gRPC channels are at a lower level than the clients. The gRPC channel builder is used elsewhere outside of the abstract client where there isn't necessarily always a ClientContext (`see ConfigurationUtils.loadClusterDefaults()`)